virtual_ecosystem/models/animals/animal_model.py

```diff
@@ -245,3 +299,41 @@ def calculate_litter_additions(self) -> dict[str, DataArray]:
                 dims="cell_id",
             ),
         }
+
+    def update_population_densities(self) -> None:
+        """Updates the densities for each functional group in each community."""
+
+        for community_id, community in self.communities.items():
+            for fg_name, cohorts in community.animal_cohorts.items():
+                # Initialize the population density of the functional group
+                fg_density = 0.0
+                for cohort in cohorts:
+                    # Calculate the population density for the cohort
+                    fg_density += self.calculate_density_for_cohort(cohort)
+
```

Comment:
I guess that this might get more complicated in the multi-grid case, but as it is now, this can be done simply as:

```suggestion
                fg_density = sum([c.individuals for c in cohorts]) / self.data.grid.cell_area
```

And you save one function.